utils/obsproc/Smos2Ioda.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <iostream>
+#include <netcdf>    // NOLINT (using C API)
+#include <string>
+#include <vector>
+
+#include "eckit/config/LocalConfiguration.h"
+
+#include <Eigen/Dense>    // NOLINT
+
+#include "ioda/Group.h"
+#include "ioda/ObsGroup.h"
+
+#include "NetCDFToIodaConverter.h"
+
+namespace gdasapp {
+
+  class Smos2Ioda : public NetCDFToIodaConverter {
+   public:
+    explicit Smos2Ioda(const eckit::Configuration & fullConfig)
+    : NetCDFToIodaConverter(fullConfig) {
+      variable_ = "Salinity";
+    }
+
+    // Read netcdf file and populate iodaVars
+    gdasapp::IodaVars providerToIodaVars(const std::string fileName) final {
+      oops::Log::info() << "Processing files provided by SMOS" << std::endl;
+
+      // Open the NetCDF file in read-only mode
+      netCDF::NcFile ncFile(fileName, netCDF::NcFile::read);
+
+      // Get number of obs
+      int nobs = ncFile.getDim("n_grid_points").getSize();
+
+      // Set the int metadata names
+      // TODO(AFE): add other metadata in form of
+      // std::vector<std::string> intMetadataNames = {"pass", "cycle", "mission"};
```

Comment:
both are part  of the TODO's @ShastriPaturi , seems fine to me as a comment, more so since we know it's work in progress and the next feature will be the addition of metadata.